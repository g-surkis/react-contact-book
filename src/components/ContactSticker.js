import React, { Component } from "react";
import PropTypes from "prop-types";

import { DragSource, DropTarget } from "react-dnd";
import flow from "lodash/flow";

const cardSource = {

  beginDrag(props, monitor, component) {
    props.itemDragged(props.contact);
    return props.contact;
  }
};

const cardTarget = { 
 
  hover(props, monitor, component) {
    const dragIndex = monitor.getItem().index;
    const hoverIndex = props.index;

    if (dragIndex === hoverIndex) {
      return;
    }
    props.moveCard(hoverIndex);

    monitor.getItem().index = hoverIndex;
  }
};


class ContactSticker extends Component {
  componentWillReceiveProps(nextProps) {
    console.log(!this.props.isDragging && nextProps.isOver);
    if (!this.props.isDragging && nextProps.isOver ) {

      const dragElement = document.getElementById("card" + this.props.index);
      console.log(dragElement);
       dragElement.style.opacity = 1;
    }
  }

  compo

  render() {
    const {
      isDragging,
      connectDragSource,
      connectDropTarget,
      isOver
    } = this.props;
    const opacity = (isDragging || isOver)  ? 0 : 1 ;


    return connectDragSource(
      connectDropTarget( 
        <figure
          className="contact_item"
          id = {"card" + this.props.index}
          onClick={this.props.click}
          style={{
            opacity
          }}
        >
          <h4>
            {this.props.contact.firstName} {this.props.contact.lastName}
          </h4>
          <p>{this.props.contact.phone}</p>
          <p>{this.props.contact.email}</p>
        </figure>
      )
    );
  }
}

export default flow(
  DragSource("card", cardSource, (connect, monitor) => ({
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  })),
  DropTarget("card", cardTarget, (connect, monitor) => ({
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver({ shallow: true })
  }))
)(ContactSticker);

ContactSticker.propTypes = {
  click: PropTypes.func,
  connectDragSource: PropTypes.func,
  connectDropTarget: PropTypes.func,
  contactKey: PropTypes.number,
  index: PropTypes.number,
  isDragging: PropTypes.bool,
  itemDragged: PropTypes.func,
  label: PropTypes.string,
  moveCard: PropTypes.func
};
