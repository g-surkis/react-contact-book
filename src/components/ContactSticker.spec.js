import React from "react";
import { shallow, mount } from "enzyme";
import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store';

import ContactSticker from "./ContactSticker";

describe('>>>C O N T A C T  S T I C K E R --- REACT-REDUX (Shallow + passing the {store} directly)',()=>{
    const initialState = { contactList: [{}] };
    const mockStore = configureStore();
    let store,container;

    beforeEach(()=>{
        store = mockStore(initialState);
        container = shallow(<ContactSticker store={store} /> );
    })



    describe('>>>C O N T A C T  S T I C K E R  --- Snapshot',()=>{
        it('+++capturing Snapshot of ContactList', () => {
            const renderedValue =  renderer.create(<ContactSticker  store={store}/>).toJSON()
            expect(renderedValue).toMatchSnapshot();
        });
    });
})