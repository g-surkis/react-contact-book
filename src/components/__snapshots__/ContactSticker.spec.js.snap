// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>C O N T A C T  S T I C K E R --- REACT-REDUX (Shallow + passing the {store} directly) >>>C O N T A C T  S T I C K E R  --- Snapshot +++capturing Snapshot of ContactList 1`] = `null`;
